src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Utility/GetPackageReferenceUtility.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.ProjectModel;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace NuGet.PackageManagement.VisualStudio.Utility
+{
+    internal static class GetPackageReferenceUtility
+    {
+        /// <summary>
+        /// Compares the project and the assets files returning the installed package.
+        /// Assets information can be null and returns the package from the project files.
+        /// Checks if package already exists in the project and return it, otherwise update the project installed packages.
+        /// </summary>
+        /// <param name="projectLibrary">Library from the project file.</param>
+        /// <param name="targetFramework">Target framework from the project file.</param>
+        /// <param name="assetsTargetFrameworkInformation">Target framework information from the assets file.</param>
+        /// <param name="targets"></param>
+        /// <param name="installedPackages">Installed packages information from the project.</param>
+        internal static PackageIdentity UpdateResolvedVersion(LibraryDependency projectLibrary, NuGetFramework targetFramework, TargetFrameworkInformation assetsTargetFrameworkInformation, IList<LockFileTarget> targets, Dictionary<string, ProjectInstalledPackage> installedPackages)
```

Comment:
These comments look good to me.

Reply:
Some text for the `targets` parameter would be nice.